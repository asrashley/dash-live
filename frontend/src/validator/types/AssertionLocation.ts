export interface AssertionLocation {
    filename: string;
    line: number;
}
