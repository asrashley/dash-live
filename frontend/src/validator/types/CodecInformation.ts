export interface CodecDetails {
    label: string;
    error?: string;
    details: string[];
}

export interface CodecInformation {
    codec: string;
    error?: string;
    details: CodecDetails[];
}
