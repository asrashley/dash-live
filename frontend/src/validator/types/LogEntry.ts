export type LogEntry = {
    id: number;
    level: 'error' | 'info' | 'debug';
    text: string;
};
