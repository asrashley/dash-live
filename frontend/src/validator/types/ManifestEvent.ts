
export interface ManifestEvent {
    text: string[];
}