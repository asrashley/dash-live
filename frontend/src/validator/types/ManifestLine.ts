export interface ManifestLine {
    line: number;
    text: string;
    hasError: boolean;
    errors: string[];
}
