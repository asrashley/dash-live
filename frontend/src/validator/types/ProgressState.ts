export interface ProgressState {
    currentValue?: number;
    minValue: number;
    maxValue: number;
    text: string;
    finished?: boolean;
    error?: boolean;
 }
