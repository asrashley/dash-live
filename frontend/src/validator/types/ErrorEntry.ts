import { AssertionLocation } from "./AssertionLocation";

export interface ErrorEntry {
    assertion: AssertionLocation;
    location: [number, number];
    clause?: string;
    msg: string;
}