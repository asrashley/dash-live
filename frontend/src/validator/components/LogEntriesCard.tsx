import { UseValidatorWebsocketHook } from "../hooks/useValidatorWebsocket";
import { LogEntry } from "../types/LogEntry";

function LogLine({ text, level }: LogEntry) {
  return <p className={level}>{text}</p>;
}

export interface LogEntriesCardProps {
  log: UseValidatorWebsocketHook["log"];
}

export function LogEntriesCard({ log }: LogEntriesCardProps) {
  return (
    <div className="card results">
      {log.value.map((msg) => (
        <LogLine key={msg.id} {...msg} />
      ))}
    </div>
  );
}
