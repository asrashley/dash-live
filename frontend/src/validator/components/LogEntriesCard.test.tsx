import { describe, expect, test } from "vitest";
import { signal } from "@preact/signals-core";

import { LogEntry } from "../types/LogEntry";
import { renderWithProviders } from "../../test/renderWithProviders";
import { LogEntriesCard } from "./LogEntriesCard";

describe("LogEntriesCard component", () => {
  const logItems: LogEntry[] = [
    {
      level: "info",
      text: "Prefetching media files required before validation can start",
      id: 1,
    },
    { level: "info", text: "Starting stream validation...", id: 2 },
    {
      level: "info",
      text: "Prefetching media files required before validation can start",
      id: 3,
    },
    { level: "info", text: "Validation complete after 11.0 seconds", id: 4 },
  ];
  const log = signal<LogEntry[]>([]);

  test("matches snapshot", () => {
    log.value = structuredClone(logItems);
    const { asFragment, getAllByText } = renderWithProviders(<LogEntriesCard log={log} />);
    logItems.forEach(({text}) => getAllByText(text));
    expect(asFragment()).toMatchSnapshot();
  });
});
