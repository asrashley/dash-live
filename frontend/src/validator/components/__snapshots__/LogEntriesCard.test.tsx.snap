// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LogEntriesCard component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card results"
  >
    <p
      class="info"
    >
      Prefetching media files required before validation can start
    </p>
    <p
      class="info"
    >
      Starting stream validation...
    </p>
    <p
      class="info"
    >
      Prefetching media files required before validation can start
    </p>
    <p
      class="info"
    >
      Validation complete after 11.0 seconds
    </p>
  </div>
</DocumentFragment>
`;
